runtime/ms-rest/lib/webResource.js

```diff
@@ -293,4 +298,244 @@ WebResource.prototype.pipeInput = function(inputStream, destStream) {
+          } else {
+            url = url.replace(item, encodeURIComponent(pathParam.value));
+          }
+        }
+      });
+    }
+    this.url = url;
+  }
+
+  //append query parameters to the url if they are provided. They can be provided with pathTemplate or url option.
+  if (options.queryParameters) {
+    if (typeof options.queryParameters !== 'object') {
+      throw new Error('options.queryParameters must be of type object. It should be a JSON object ' +
+        'of \'query-parameter-name\' as the key and the \'query-parameter-value\' as the value. ' +
+        'The \'query-parameter-value\' may be a string or an object of the form { value: \'query-parameter-value\', skipUrlEncoding: true }.');
+    }
+    //append question mark if it is not present in the url
+    if (this.url && this.url.indexOf('?') === -1) {
+      this.url += '?';
+    }
+    //construct queryString
+    var queryParams = [];
+    var queryParameters = options.queryParameters;
+    //We need to populate this.query as a dictionary if the request is being used for Sway's validateRequest(). 
+    this.query = {};
+    for (var queryParamName in queryParameters) {
+      var queryParam = queryParameters[queryParamName];
+      if (queryParam) {
+        if (typeof queryParam === 'string') {
+          queryParams.push(queryParamName + '=' + encodeURIComponent(queryParam));
+          this.query[queryParamName] = encodeURIComponent(queryParam);
+        }
+        if (typeof queryParam === 'object') {
+          if (!queryParam.value) {
+            throw new Error('options.queryParameters[' + queryParamName + '] is of type \'object\' but it does not contain a \'value\' property.');
+          }
+          if (queryParam.skipUrlEncoding) {
+            queryParams.push(queryParamName + '=' + queryParameters[queryParamName]);
+            this.query[queryParamName] = queryParameters[queryParamName];
+          } else {
+            queryParams.push(queryParamName + '=' + encodeURIComponent(queryParameters[queryParamName]));
+            this.query[queryParamName] = encodeURIComponent(queryParameters[queryParamName]);
+          }
+        }
+      }
+    }//end-of-for
+    //append the queryString
+    this.url += queryParams.join('&');
+  }
+
+  //add headers to the request if they are provided
+  if (options.headers) {
+    var headers = options.headers;
+    for (var headerName in headers) {
+      if (headers.hasOwnProperty(headerName)) {
+        this.headers[headerName] = headers[headerName];
+      }
+    }
+  }
+  //ensure accept-language is set correctly
+  if (!this.headers['accept-language']) {
+    this.headers['accept-language'] = 'en-US';
+  }
+  //ensure the request-id is set correctly
+  if (!this.headers['x-ms-client-request-id'] && !options.disableClientRequestId) {
+    this.headers['x-ms-client-request-id'] = utils.generateUuid();
+  }
+  //ensure content-type is set correctly
+  if (this.headers['Content-Type'] &&
```

Comment:
we are doing this in the all the autorest generated sdks. An example over [here](https://github.com/Azure/azure-sdk-for-node/blob/master/lib/services/storageManagement2/lib/operations/storageAccounts.js#L121). I think it is a standard practice.